### PR TITLE
refactor: Simplify error classes

### PR DESCRIPTION
**Pull request by gregor**

### CODE REVIEW

File: app/script/auth/AuthError.js [hunk 1]
```diff
@@ -23,37 +23,33 @@ window.z = window.z || {};
...
-      case AuthError.TYPE.MULTIPLE_TABS:
-        this.message = 'Cannot open in multiple tabs simultaneously';
-        break;
-      case AuthError.TYPE.PRIVATE_MODE:
-        this.message = 'Unsupported Private Mode';
-        break;
-      default:
-        this.message = 'Unknown AuthError';
-    }
+    this.message = message || AuthError.MESSAGE[this.type] || AuthError.MESSAGE.UNKNOWN;
```
Comment (atomrc) [inline app/script/auth/AuthError.js]: I see this pattern quite some times in this PR. Maybe having a super class that does that could be a good idea. 
How about a `WireError` and make all of those inherit from it?
WDYT?

Comment (gregor) [reply app/script/auth/AuthError.js]: Yeah. Sounds like a good idea.

File: app/script/error/AccessTokenError.js [hunk 2]
```diff
@@ -24,8 +24,7 @@ window.z.error = z.error || {};
 
 z.error.AccessTokenError = class AccessTokenError extends z.error.BaseError {
   constructor(type, message) {
-    const errorName = 'AccessTokenError';
-    super(errorName, type, message);
+    super(type, message);
```
Comment (atomrc) [inline app/script/error/AccessTokenError.js]: In those cases, you can remove the `constructor` at all

File: app/script/error/BaseError.js [hunk 3]
```diff
@@ -28,7 +28,12 @@ z.error.BaseError = class BaseError extends Error {
 
     this.name = this.constructor.name;
     this.stack = new Error().stack;
-    this.type = type || BaseError.TYPE.UNKNOWN;
+
+    const specificTypes = Object.values(z.error[this.name].TYPE);
+    const knownTypes = Object.vales(BaseError.TYPE).concat(specificTypes);
```
Comment (atomrc) [inline app/script/error/BaseError.js]: Typo : Object.vale

**Pull request merged**